lib/texutil.js

```diff
@@ -42,11 +42,128 @@ module.exports.latex_function_names = arr2set([
     "\\sin", "\\sinh", "\\sup", "\\tan", "\\tanh"
 ]);
 
+function findBracketSections(string) {
+    var sections = [[""]];
+    var char;
+    var level = 0;
+    for (var i = 0; i < string.length; i++) {
+        char = string.charAt(i);
+        if (char === "}" || char === "]") {
+            level--;
+            if (level === 0) {
+                sections.push([""]);
+            }
+            else {
+                sections[sections.length - 1][0] += char;
+            }
+        }
+        else if (char === "{" || char === "[") {
+            if (level !== 0) {
+                sections[sections.length - 1][0] += char;
+            }
+            else {
+                sections[sections.length - 1].push(char);
+            }
+            level++;
+        }
+        else {
+            sections[sections.length - 1][0] += char;
+        }
+    }
+    sections = sections.slice(0, sections.length - 1);
+    return sections;
+}
+
+function findFirst(string, subs, start) {
+    var min = -1;
+    var index;
+    var first = subs[0];
+    for (var i = 0; i < subs.length; i++) {
+        index = string.indexOf(subs[i], start);
+        if ((index !== -1) && (index < min || min === -1)) {
+            min = index;
+            first = subs[i];
+        }
+    }
+    return [min, first];
+}
+
+function parseNewCommand(definition, command, cmds) {
+    definition = definition.substring(command.length);
+    var sections = findBracketSections(definition);
+    if (sections.length === 0) {
+        var commands = definition.match(/\\[a-zA-Z]+/g);
+        cmds[commands[0]] = commands[1];
+    }
+    else if (sections.length === 1) {
+        var cmd = definition.replaceAll(sections[0][0], "").match(/\\[a-zA-Z]+/)[0];
+        if (definition.indexOf(cmd) === 0) {
+            cmds[cmd] = sections[0][0];
+        }
+        else {
+            cmds[sections[0][0]] = cmd;
+        }
+    }
+    else if (sections.length === 2) {
+        if (sections[0][1] !== "[") {
+            cmds[sections[0][0]] = sections[1][0];
+        }
+    }
+}
+
+function bracketBalance(string) {
```

Comment:
The idea of unit tests is that each small functional unit has the corresponding unit tests.
Maybe you can a few tests of each of the new method you introduce.